test/test_http20.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+"""
+test_http20.py
+~~~~~~~~~~~~~~
+
+Unit tests for hyper's HTTP/2.0 implementation.
+"""
+import pytest
+from mock import patch
+
+from server import SocketLevelTest
+
+
+class TestHTTP20Connection(SocketLevelTest):
+    h2 = True
+
+    def test_useful_error_with_no_protocol(self):
+        self.set_up()
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+            sock.close()
+
+        self._start_server(socket_handler)
+        conn = self.get_connection()
+
+        with patch('hyper.http20.connection.wrap_socket') as mock:
+            mock.return_value = (None, None)
+            with pytest.raises(AssertionError) as exc_info:
+                conn.connect()
+                assert (
+                    "No suitable protocol found, check your OpenSSL version"
+                    in
+                    str(exc_info)
+                )
```

Comment:
You can pull this assertion out of the context manager, which you will in fact need to do to get it to run because the `AssertionError` that gets thrown terminates the execution of this block.